### PR TITLE
Bluetooth: Controller: Offset Coded PHY scan window

### PR DESCRIPTION
**Pull request by cvinayak**
Offset the Coded PHY scan window by 1M PHY scan window
duration so that when 1M PHY scan window is less than
scan interval then Coded PHY scan window can utilize the
radio thereafter for its scan window duration. For
example, with 20 ms interval and 10 ms window there can
be 10 ms 1M PHY followed by 10 ms of Coded PHY scanning.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_scan.c [hunk 1]
```diff
@@ -435,12 +437,31 @@ uint8_t ull_scan_enable(struct ll_scan_set *scan)
...
+	handle = ull_scan_handle_get(scan);
+
+	if (false) {
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_CTLR_PHY_CODED)
+	} else if (handle == SCAN_HANDLE_PHY_CODED) {
+		struct ll_scan_set *scan_1m;
+
+		scan_1m = ull_scan_set_get(SCAN_HANDLE_1M);
+		if (scan_1m->lll.phy & PHY_1M) {
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_scan.c:449]: `struct ull_scan_set` is common context structure one each for `PHY_1M` and `PHY_CODED` (and scalable for future if new PHY are added). `scan_1m->lll.phy & PHY_1M` and `scan_coded->lll.phy & PHY_CODED` indicate if scanning is enable on either or both PHYs. Hence, the `if` is needed to check if the scanning is enabled.

[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/controller/ll_sw/ull_scan.c:449]: I see. That's pretty unclear from the code. Can we improve on that somehow?

[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_scan.c]: Added a macro to check if scanning on PHY_1M and/or PHY_CODED is enabled.

**Review by Thalley — DISMISSED** [review]

**Comment by cvinayak** [comment]
@Thalley sorry, had to fix compliance check failure, please approve after CI passes, thanks.

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**